File: Models/campbell_SoilTemp/Campbell/Campbell/src/bioma/Campbell/model_SoilTempCampbellState.cs

using System;
using System.Collections.Generic;
using CRA.ModelLayer.Core;
using System.Reflection;
using CRA.ModelLayer.ParametersManagement;   

namespace model_SoilTempCampbell.DomainClass
{
    public class model_SoilTempCampbellState : ICloneable, IDomainClass
    {
        private double[] _THICKApsim;
        private double[] _DEPTHApsim;
        private double[] _BDApsim;
        private double[] _CLAYApsim;
        private double[] _SWApsim;
        private double[] _soilTemp;
        private double[] _newTemperature;
        private double[] _minSoilTemp;
        private double[] _maxSoilTemp;
        private double[] _aveSoilTemp;
        private double[] _morningSoilTemp;
        private double[] _thermalCondPar1;
        private double[] _thermalCondPar2;
        private double[] _thermalCondPar3;
        private double[] _thermalCondPar4;
        private double[] _thermalConductivity;
        private double[] _thermalConductance;
        private double[] _heatStorage;
        private double[] _volSpecHeatSoil;
        private double _maxTempYesterday;
        private double _minTempYesterday;
        private double[] _SLCARBApsim;
        private double[] _SLROCKApsim;
        private double[] _SLSILTApsim;
        private double[] _SLSANDApsim;
        private double __boundaryLayerConductance;
        private ParametersIO _parametersIO;

        public model_SoilTempCampbellState()
        {
            _parametersIO = new ParametersIO(this);
        }

        public model_SoilTempCampbellState(model_SoilTempCampbellState toCopy, bool copyAll) // copy constructor 
        {
            if (copyAll)
            {
                        THICKApsim = new double[toCopy.THICKApsim.Length];
            for (int i = 0; i < toCopy.THICKApsim.Length; i++)
                { THICKApsim[i] = toCopy.THICKApsim[i]; }
    
                        DEPTHApsim = new double[toCopy.DEPTHApsim.Length];
            for (int i = 0; i < toCopy.DEPTHApsim.Length; i++)
                { DEPTHApsim[i] = toCopy.DEPTHApsim[i]; }
    
                        BDApsim = new double[toCopy.BDApsim.Length];
            for (int i = 0; i < toCopy.BDApsim.Length; i++)
                { BDApsim[i] = toCopy.BDApsim[i]; }
    
                        CLAYApsim = new double[toCopy.CLAYApsim.Length];
            for (int i = 0; i < toCopy.CLAYApsim.Length; i++)
                { CLAYApsim[i] = toCopy.CLAYApsim[i]; }
    
                        SWApsim = new double[toCopy.SWApsim.Length];
            for (int i = 0; i < toCopy.SWApsim.Length; i++)
                { SWApsim[i] = toCopy.SWApsim[i]; }
    
                        soilTemp = new double[toCopy.soilTemp.Length];
            for (int i = 0; i < toCopy.soilTemp.Length; i++)
                { soilTemp[i] = toCopy.soilTemp[i]; }
    
                        newTemperature = new double[toCopy.newTemperature.Length];
            for (int i = 0; i < toCopy.newTemperature.Length; i++)
                { newTemperature[i] = toCopy.newTemperature[i]; }
    
                        minSoilTemp = new double[toCopy.minSoilTemp.Length];
            for (int i = 0; i < toCopy.minSoilTemp.Length; i++)
                { minSoilTemp[i] = toCopy.minSoilTemp[i]; }
    
                        maxSoilTemp = new double[toCopy.maxSoilTemp.Length];
            for (int i = 0; i < toCopy.maxSoilTemp.Length; i++)
                { maxSoilTemp[i] = toCopy.maxSoilTemp[i]; }
    
                        aveSoilTemp = new double[toCopy.aveSoilTemp.Length];
            for (int i = 0; i < toCopy.aveSoilTemp.Length; i++)
                { aveSoilTemp[i] = toCopy.aveSoilTemp[i]; }
    
                        morningSoilTemp = new double[toCopy.morningSoilTemp.Length];
            for (int i = 0; i < toCopy.morningSoilTemp.Length; i++)
                { morningSoilTemp[i] = toCopy.morningSoilTemp[i]; }
    
                        thermalCondPar1 = new double[toCopy.thermalCondPar1.Length];
            for (int i = 0; i < toCopy.thermalCondPar1.Length; i++)
                { thermalCondPar1[i] = toCopy.thermalCondPar1[i]; }
    
                        thermalCondPar2 = new double[toCopy.thermalCondPar2.Length];
            for (int i = 0; i < toCopy.thermalCondPar2.Length; i++)
                { thermalCondPar2[i] = toCopy.thermalCondPar2[i]; }
    
                        thermalCondPar3 = new double[toCopy.thermalCondPar3.Length];
            for (int i = 0; i < toCopy.thermalCondPar3.Length; i++)
                { thermalCondPar3[i] = toCopy.thermalCondPar3[i]; }
    
                        thermalCondPar4 = new double[toCopy.thermalCondPar4.Length];
            for (int i = 0; i < toCopy.thermalCondPar4.Length; i++)
                { thermalCondPar4[i] = toCopy.thermalCondPar4[i]; }
    
                        thermalConductivity = new double[toCopy.thermalConductivity.Length];
            for (int i = 0; i < toCopy.thermalConductivity.Length; i++)
                { thermalConductivity[i] = toCopy.thermalConductivity[i]; }
    
                        thermalConductance = new double[toCopy.thermalConductance.Length];
            for (int i = 0; i < toCopy.thermalConductance.Length; i++)
                { thermalConductance[i] = toCopy.thermalConductance[i]; }
    
                        heatStorage = new double[toCopy.heatStorage.Length];
            for (int i = 0; i < toCopy.heatStorage.Length; i++)
                { heatStorage[i] = toCopy.heatStorage[i]; }
    
                        volSpecHeatSoil = new double[toCopy.volSpecHeatSoil.Length];
            for (int i = 0; i < toCopy.volSpecHeatSoil.Length; i++)
                { volSpecHeatSoil[i] = toCopy.volSpecHeatSoil[i]; }
    
                        maxTempYesterday = toCopy.maxTempYesterday;
                        minTempYesterday = toCopy.minTempYesterday;
                        SLCARBApsim = new double[toCopy.SLCARBApsim.Length];
            for (int i = 0; i < toCopy.SLCARBApsim.Length; i++)
                { SLCARBApsim[i] = toCopy.SLCARBApsim[i]; }
    
                        SLROCKApsim = new double[toCopy.SLROCKApsim.Length];
            for (int i = 0; i < toCopy.SLROCKApsim.Length; i++)
                { SLROCKApsim[i] = toCopy.SLROCKApsim[i]; }
    
                        SLSILTApsim = new double[toCopy.SLSILTApsim.Length];
            for (int i = 0; i < toCopy.SLSILTApsim.Length; i++)
                { SLSILTApsim[i] = toCopy.SLSILTApsim[i]; }
    
                        SLSANDApsim = new double[toCopy.SLSANDApsim.Length];
            for (int i = 0; i < toCopy.SLSANDApsim.Length; i++)
                { SLSANDApsim[i] = toCopy.SLSANDApsim[i]; }
    
                        _boundaryLayerConductance = toCopy._boundaryLayerConductance;
                    }
                }

                public double[] THICKApsim
    {
        get { return this._THICKApsim; }
        set { this._THICKApsim= value; } 
    }
                public double[] DEPTHApsim
    {
        get { return this._DEPTHApsim; }
        set { this._DEPTHApsim= value; } 
    }
                public double[] BDApsim
    {
        get { return this._BDApsim; }
        set { this._BDApsim= value; } 
    }
                public double[] CLAYApsim
    {
        get { return this._CLAYApsim; }
        set { this._CLAYApsim= value; } 
    }
                public double[] SWApsim
    {
        get { return this._SWApsim; }
        set { this._SWApsim= value; } 
    }
                public double[] soilTemp
    {
        get { return this._soilTemp; }
        set { this._soilTemp= value; } 
    }
                public double[] newTemperature
    {
        get { return this._newTemperature; }
        set { this._newTemperature= value; } 
    }
                public double[] minSoilTemp
    {
        get { return this._minSoilTemp; }
        set { this._minSoilTemp= value; } 
    }
                public double[] maxSoilTemp
    {
        get { return this._maxSoilTemp; }
        set { this._maxSoilTemp= value; } 
    }
                public double[] aveSoilTemp
    {
        get { return this._aveSoilTemp; }
        set { this._aveSoilTemp= value; } 
    }
                public double[] morningSoilTemp
    {
        get { return this._morningSoilTemp; }
        set { this._morningSoilTemp= value; } 
    }
                public double[] thermalCondPar1
    {
        get { return this._thermalCondPar1; }
        set { this._thermalCondPar1= value; } 
    }
                public double[] thermalCondPar2
    {
        get { return this._thermalCondPar2; }
        set { this._thermalCondPar2= value; } 
    }
                public double[] thermalCondPar3
    {
        get { return this._thermalCondPar3; }
        set { this._thermalCondPar3= value; } 
    }
                public double[] thermalCondPar4
    {
        get { return this._thermalCondPar4; }
        set { this._thermalCondPar4= value; } 
    }
                public double[] thermalConductivity
    {
        get { return this._thermalConductivity; }
        set { this._thermalConductivity= value; } 
    }
                public double[] thermalConductance
    {
        get { return this._thermalConductance; }
        set { this._thermalConductance= value; } 
    }
                public double[] heatStorage
    {
        get { return this._heatStorage; }
        set { this._heatStorage= value; } 
    }
                public double[] volSpecHeatSoil
    {
        get { return this._volSpecHeatSoil; }
        set { this._volSpecHeatSoil= value; } 
    }
                public double maxTempYesterday
    {
        get { return this._maxTempYesterday; }
        set { this._maxTempYesterday= value; } 
    }
                public double minTempYesterday
    {
        get { return this._minTempYesterday; }
        set { this._minTempYesterday= value; } 
    }
                public double[] SLCARBApsim
    {
        get { return this._SLCARBApsim; }
        set { this._SLCARBApsim= value; } 
    }
                public double[] SLROCKApsim
    {
        get { return this._SLROCKApsim; }
        set { this._SLROCKApsim= value; } 
    }
                public double[] SLSILTApsim
    {
        get { return this._SLSILTApsim; }
        set { this._SLSILTApsim= value; } 
    }
                public double[] SLSANDApsim
    {
        get { return this._SLSANDApsim; }
        set { this._SLSANDApsim= value; } 
    }
                public double _boundaryLayerConductance
    {
        get { return this.__boundaryLayerConductance; }
        set { this.__boundaryLayerConductance= value; } 
    }

                public string Description
                {
                    get { return "model_SoilTempCampbellState of the component";}
                }

                public string URL
                {
                    get { return "http://" ;}
                }

                public virtual IDictionary<string, PropertyInfo> PropertiesDescription
                {
                    get { return _parametersIO.GetCachedProperties(typeof(IDomainClass));}
                }

                public virtual Boolean ClearValues()
                {
                     _THICKApsim = new double[NLAYR];
                     _DEPTHApsim = new double[NLAYR];
                     _BDApsim = new double[NLAYR];
                     _CLAYApsim = new double[NLAYR];
                     _SWApsim = new double[NLAYR];
                     _soilTemp = new double[NLAYR];
                     _newTemperature = new double[NLAYR];
                     _minSoilTemp = new double[NLAYR];
                     _maxSoilTemp = new double[NLAYR];
                     _aveSoilTemp = new double[NLAYR];
                     _morningSoilTemp = new double[NLAYR];
                     _thermalCondPar1 = new double[NLAYR];
                     _thermalCondPar2 = new double[NLAYR];
                     _thermalCondPar3 = new double[NLAYR];
                     _thermalCondPar4 = new double[NLAYR];
                     _thermalConductivity = new double[NLAYR];
                     _thermalConductance = new double[NLAYR];
                     _heatStorage = new double[NLAYR];
                     _volSpecHeatSoil = new double[NLAYR];
                     _maxTempYesterday = default(double);
                     _minTempYesterday = default(double);
                     _SLCARBApsim = new double[NLAYR];
                     _SLROCKApsim = new double[NLAYR];
                     _SLSILTApsim = new double[NLAYR];
                     _SLSANDApsim = new double[NLAYR];
                     __boundaryLayerConductance = default(double);
                    return true;
                }

                public virtual Object Clone()
                {
                    IDomainClass myclass = (IDomainClass) this.MemberwiseClone();
                    _parametersIO.PopulateClonedCopy(myclass);
                    return myclass;
                }
            }
        }

File: Models/campbell_SoilTemp/Campbell/src/java/Campbell/model_SoilTempCampbellState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class Model_SoilTempCampbellState
{
    private List<Double> THICKApsim;
    private List<Double> DEPTHApsim;
    private List<Double> BDApsim;
    private List<Double> CLAYApsim;
    private List<Double> SWApsim;
    private List<Double> soilTemp;
    private List<Double> newTemperature;
    private List<Double> minSoilTemp;
    private List<Double> maxSoilTemp;
    private List<Double> aveSoilTemp;
    private List<Double> morningSoilTemp;
    private List<Double> thermalCondPar1;
    private List<Double> thermalCondPar2;
    private List<Double> thermalCondPar3;
    private List<Double> thermalCondPar4;
    private List<Double> thermalConductivity;
    private List<Double> thermalConductance;
    private List<Double> heatStorage;
    private List<Double> volSpecHeatSoil;
    private double maxTempYesterday;
    private double minTempYesterday;
    private List<Double> SLCARBApsim;
    private List<Double> SLROCKApsim;
    private List<Double> SLSILTApsim;
    private List<Double> SLSANDApsim;
    private double _boundaryLayerConductance;
    
    public Model_SoilTempCampbellState() { }
    
    public Model_SoilTempCampbellState(Model_SoilTempCampbellState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            List <Double> _THICKApsim = new ArrayList<>();
            for (Double c : toCopy.getTHICKApsim())
            {
                _THICKApsim.add(c);
            }
            this.THICKApsim = _THICKApsim;
            List <Double> _DEPTHApsim = new ArrayList<>();
            for (Double c : toCopy.getDEPTHApsim())
            {
                _DEPTHApsim.add(c);
            }
            this.DEPTHApsim = _DEPTHApsim;
            List <Double> _BDApsim = new ArrayList<>();
            for (Double c : toCopy.getBDApsim())
            {
                _BDApsim.add(c);
            }
            this.BDApsim = _BDApsim;
            List <Double> _CLAYApsim = new ArrayList<>();
            for (Double c : toCopy.getCLAYApsim())
            {
                _CLAYApsim.add(c);
            }
            this.CLAYApsim = _CLAYApsim;
            List <Double> _SWApsim = new ArrayList<>();
            for (Double c : toCopy.getSWApsim())
            {
                _SWApsim.add(c);
            }
            this.SWApsim = _SWApsim;
            List <Double> _soilTemp = new ArrayList<>();
            for (Double c : toCopy.getsoilTemp())
            {
                _soilTemp.add(c);
            }
            this.soilTemp = _soilTemp;
            List <Double> _newTemperature = new ArrayList<>();
            for (Double c : toCopy.getnewTemperature())
            {
                _newTemperature.add(c);
            }
            this.newTemperature = _newTemperature;
            List <Double> _minSoilTemp = new ArrayList<>();
            for (Double c : toCopy.getminSoilTemp())
            {
                _minSoilTemp.add(c);
            }
            this.minSoilTemp = _minSoilTemp;
            List <Double> _maxSoilTemp = new ArrayList<>();
            for (Double c : toCopy.getmaxSoilTemp())
            {
                _maxSoilTemp.add(c);
            }
            this.maxSoilTemp = _maxSoilTemp;
            List <Double> _aveSoilTemp = new ArrayList<>();
            for (Double c : toCopy.getaveSoilTemp())
            {
                _aveSoilTemp.add(c);
            }
            this.aveSoilTemp = _aveSoilTemp;
            List <Double> _morningSoilTemp = new ArrayList<>();
            for (Double c : toCopy.getmorningSoilTemp())
            {
                _morningSoilTemp.add(c);
            }
            this.morningSoilTemp = _morningSoilTemp;
            List <Double> _thermalCondPar1 = new ArrayList<>();
            for (Double c : toCopy.getthermalCondPar1())
            {
                _thermalCondPar1.add(c);
            }
            this.thermalCondPar1 = _thermalCondPar1;
            List <Double> _thermalCondPar2 = new ArrayList<>();
            for (Double c : toCopy.getthermalCondPar2())
            {
                _thermalCondPar2.add(c);
            }
            this.thermalCondPar2 = _thermalCondPar2;
            List <Double> _thermalCondPar3 = new ArrayList<>();
            for (Double c : toCopy.getthermalCondPar3())
            {
                _thermalCondPar3.add(c);
            }
            this.thermalCondPar3 = _thermalCondPar3;
            List <Double> _thermalCondPar4 = new ArrayList<>();
            for (Double c : toCopy.getthermalCondPar4())
            {
                _thermalCondPar4.add(c);
            }
            this.thermalCondPar4 = _thermalCondPar4;
            List <Double> _thermalConductivity = new ArrayList<>();
            for (Double c : toCopy.getthermalConductivity())
            {
                _thermalConductivity.add(c);
            }
            this.thermalConductivity = _thermalConductivity;
            List <Double> _thermalConductance = new ArrayList<>();
            for (Double c : toCopy.getthermalConductance())
            {
                _thermalConductance.add(c);
            }
            this.thermalConductance = _thermalConductance;
            List <Double> _heatStorage = new ArrayList<>();
            for (Double c : toCopy.getheatStorage())
            {
                _heatStorage.add(c);
            }
            this.heatStorage = _heatStorage;
            List <Double> _volSpecHeatSoil = new ArrayList<>();
            for (Double c : toCopy.getvolSpecHeatSoil())
            {
                _volSpecHeatSoil.add(c);
            }
            this.volSpecHeatSoil = _volSpecHeatSoil;
            this.maxTempYesterday = toCopy.getmaxTempYesterday();
            this.minTempYesterday = toCopy.getminTempYesterday();
            List <Double> _SLCARBApsim = new ArrayList<>();
            for (Double c : toCopy.getSLCARBApsim())
            {
                _SLCARBApsim.add(c);
            }
            this.SLCARBApsim = _SLCARBApsim;
            List <Double> _SLROCKApsim = new ArrayList<>();
            for (Double c : toCopy.getSLROCKApsim())
            {
                _SLROCKApsim.add(c);
            }
            this.SLROCKApsim = _SLROCKApsim;
            List <Double> _SLSILTApsim = new ArrayList<>();
            for (Double c : toCopy.getSLSILTApsim())
            {
                _SLSILTApsim.add(c);
            }
            this.SLSILTApsim = _SLSILTApsim;
            List <Double> _SLSANDApsim = new ArrayList<>();
            for (Double c : toCopy.getSLSANDApsim())
            {
                _SLSANDApsim.add(c);
            }
            this.SLSANDApsim = _SLSANDApsim;
            this._boundaryLayerConductance = toCopy.get_boundaryLayerConductance();
        }
    }
    public List<Double> getTHICKApsim()
    { return THICKApsim; }

    public void setTHICKApsim(List<Double> _THICKApsim)
    { this.THICKApsim= _THICKApsim; } 
    
    public List<Double> getDEPTHApsim()
    { return DEPTHApsim; }

    public void setDEPTHApsim(List<Double> _DEPTHApsim)
    { this.DEPTHApsim= _DEPTHApsim; } 
    
    public List<Double> getBDApsim()
    { return BDApsim; }

    public void setBDApsim(List<Double> _BDApsim)
    { this.BDApsim= _BDApsim; } 
    
    public List<Double> getCLAYApsim()
    { return CLAYApsim; }

    public void setCLAYApsim(List<Double> _CLAYApsim)
    { this.CLAYApsim= _CLAYApsim; } 
    
    public List<Double> getSWApsim()
    { return SWApsim; }

    public void setSWApsim(List<Double> _SWApsim)
    { this.SWApsim= _SWApsim; } 
    
    public List<Double> getsoilTemp()
    { return soilTemp; }

    public void setsoilTemp(List<Double> _soilTemp)
    { this.soilTemp= _soilTemp; } 
    
    public List<Double> getnewTemperature()
    { return newTemperature; }

    public void setnewTemperature(List<Double> _newTemperature)
    { this.newTemperature= _newTemperature; } 
    
    public List<Double> getminSoilTemp()
    { return minSoilTemp; }

    public void setminSoilTemp(List<Double> _minSoilTemp)
    { this.minSoilTemp= _minSoilTemp; } 
    
    public List<Double> getmaxSoilTemp()
    { return maxSoilTemp; }

    public void setmaxSoilTemp(List<Double> _maxSoilTemp)
    { this.maxSoilTemp= _maxSoilTemp; } 
    
    public List<Double> getaveSoilTemp()
    { return aveSoilTemp; }

    public void setaveSoilTemp(List<Double> _aveSoilTemp)
    { this.aveSoilTemp= _aveSoilTemp; } 
    
    public List<Double> getmorningSoilTemp()
    { return morningSoilTemp; }

    public void setmorningSoilTemp(List<Double> _morningSoilTemp)
    { this.morningSoilTemp= _morningSoilTemp; } 
    
    public List<Double> getthermalCondPar1()
    { return thermalCondPar1; }

    public void setthermalCondPar1(List<Double> _thermalCondPar1)
    { this.thermalCondPar1= _thermalCondPar1; } 
    
    public List<Double> getthermalCondPar2()
    { return thermalCondPar2; }

    public void setthermalCondPar2(List<Double> _thermalCondPar2)
    { this.thermalCondPar2= _thermalCondPar2; } 
    
    public List<Double> getthermalCondPar3()
    { return thermalCondPar3; }

    public void setthermalCondPar3(List<Double> _thermalCondPar3)
    { this.thermalCondPar3= _thermalCondPar3; } 
    
    public List<Double> getthermalCondPar4()
    { return thermalCondPar4; }

    public void setthermalCondPar4(List<Double> _thermalCondPar4)
    { this.thermalCondPar4= _thermalCondPar4; } 
    
    public List<Double> getthermalConductivity()
    { return thermalConductivity; }

    public void setthermalConductivity(List<Double> _thermalConductivity)
    { this.thermalConductivity= _thermalConductivity; } 
    
    public List<Double> getthermalConductance()
    { return thermalConductance; }

    public void setthermalConductance(List<Double> _thermalConductance)
    { this.thermalConductance= _thermalConductance; } 
    
    public List<Double> getheatStorage()
    { return heatStorage; }

    public void setheatStorage(List<Double> _heatStorage)
    { this.heatStorage= _heatStorage; } 
    
    public List<Double> getvolSpecHeatSoil()
    { return volSpecHeatSoil; }

    public void setvolSpecHeatSoil(List<Double> _volSpecHeatSoil)
    { this.volSpecHeatSoil= _volSpecHeatSoil; } 
    
    public double getmaxTempYesterday()
    { return maxTempYesterday; }

    public void setmaxTempYesterday(double _maxTempYesterday)
    { this.maxTempYesterday= _maxTempYesterday; } 
    
    public double getminTempYesterday()
    { return minTempYesterday; }

    public void setminTempYesterday(double _minTempYesterday)
    { this.minTempYesterday= _minTempYesterday; } 
    
    public List<Double> getSLCARBApsim()
    { return SLCARBApsim; }

    public void setSLCARBApsim(List<Double> _SLCARBApsim)
    { this.SLCARBApsim= _SLCARBApsim; } 
    
    public List<Double> getSLROCKApsim()
    { return SLROCKApsim; }

    public void setSLROCKApsim(List<Double> _SLROCKApsim)
    { this.SLROCKApsim= _SLROCKApsim; } 
    
    public List<Double> getSLSILTApsim()
    { return SLSILTApsim; }

    public void setSLSILTApsim(List<Double> _SLSILTApsim)
    { this.SLSILTApsim= _SLSILTApsim; } 
    
    public List<Double> getSLSANDApsim()
    { return SLSANDApsim; }

    public void setSLSANDApsim(List<Double> _SLSANDApsim)
    { this.SLSANDApsim= _SLSANDApsim; } 
    
    public double get_boundaryLayerConductance()
    { return _boundaryLayerConductance; }

    public void set_boundaryLayerConductance(double __boundaryLayerConductance)
    { this._boundaryLayerConductance= __boundaryLayerConductance; } 
    
}